rwengine/src/objects/CharacterObject.cpp

```diff
@@ -586,15 +586,19 @@ void CharacterObject::cycleInventory(bool up) {
         currentState.currentWeapon = 0;
     } else {
         for (int j = currentState.currentWeapon - 1; j >= 0; --j) {
-            if (currentState.weapons[j].weaponId != 0) {
+            bool isFist = j == 0;
+            if (currentState.weapons[j].weaponId != 0
+                || isFist // this bit here is to let us scroll down to the fist
+                ) {
```

Comment:
nit: I believe you can remove the comment now and move it all in one line.
If you keep the comment it should probably explain that the standard check wouldn't work (presumably) because fist is weapon 0 (so the check will always fail), also you always have a fist pickup